Flaky Test_MultiUnderlayChannels: CloseRandom can close the last required underlay

## Symptom

`Test_MultiUnderlayChannels` intermittently fails in CI (passes on re-run) with:

```
not enough open underlays of type 'default', closing channel ... underlays="map[]"
--- FAIL: Test_MultiUnderlayChannels
```

## Root cause

The test sends 3000 messages on the default sender while a goroutine calls `CloseRandom` once per second. `CloseRandom` closes a **uniformly random** underlay. The channel is configured with `default {Desired: 2, Min: 1}`, so if `CloseRandom` happens to close the last remaining `default` underlay (e.g. while a re-dial of the second default is still in flight), the `default` count hits 0, the channel closes (below `Min`), and the in-flight `SendAndWaitForWire` fails - tripping the test's `NoError` assertion. It is timing-dependent: it needs the close to land before re-dial restores a second default, which is far likelier on a slow/contended CI runner than locally.

Reproduced deterministically by closing several underlays per tick on a fast ticker (forces `default` -> 0 every run), matching the CI signature.

## Fix

Make `CloseRandom` minimum-aware: only close an underlay whose type would still satisfy its constraint `Min` afterwards (reading the channel's own `constraints`). This preserves the intent - churn underlays and verify messages keep flowing while re-dial refills them - including churning a `default` underlay when two are present, but never closes the last required one, so the channel never self-closes below minimum. `priority {Min: 0}` underlays remain freely churnable (closed to zero and refilled).

Validated: deterministic root-cause repro fails before the change; with the fix the test passes 10/10 under heavy single-close churn (15ms interval) and at the normal 1s cadence.

--- a/multi_test.go
+++ b/multi_test.go
@@ -239,10 +239,25 @@ func (self *priorityChannelBase) CloseRandom(ch Channel) {
 	mc.lock.Lock()
 	defer mc.lock.Unlock()
 	underlays := mc.underlays.GetAll()
-	if len(underlays) > 0 {
-		idx := rand.Intn(len(underlays))
-		underlay := underlays[idx]
-		_ = underlay.Close()
+
+	// Only close an underlay whose type would still meet its required Min afterwards.
+	// Closing the last required underlay (e.g. default, Min: 1) would drop the channel
+	// below minimum and close it; that is not what this churn test exercises, and it
+	// raced with the random close interval, making the test flaky.
+	counts := map[string]int{}
+	for _, u := range underlays {
+		counts[GetUnderlayType(u)]++
+	}
+	var closeable []Underlay
+	for _, u := range underlays {
+		t := GetUnderlayType(u)
+		if counts[t]-1 < mc.constraints[t].Min {
+			continue
+		}
+		closeable = append(closeable, u)
+	}
+	if len(closeable) > 0 {
+		_ = closeable[rand.Intn(len(closeable))].Close()
 	}
 }
 

